go/vt/sqlparser/ast.go

```diff
@@ -499,22 +499,25 @@ func (node *Set) WalkSubtree(visit Visit) error {
 }
 
 // DDL represents a CREATE, ALTER, DROP or RENAME statement.
-// Table is set for AlterStr, DropStr, RenameStr.
+// Table is set for AlterStr, DropStr, RenameStr, ReorganizeStr
 // NewName is set for AlterStr, CreateStr, RenameStr.
 type DDL struct {
-	Action    string
-	Table     TableName
-	NewName   TableName
-	IfExists  bool
-	TableSpec *TableSpec
+	Action               string
+	Table                TableName
+	NewName              TableName
+	IfExists             bool
+	TableSpec            *TableSpec
+	PartitionName        ColIdent
```

Comment:
I was hoping we'd put these things under an *AlterSpec:

```go
type DDL struct {
  Action string
  Table TableName
  NewName TableName
  IfExists bool
  TableSpec *TableSpec
  AlterSpec *AlterSpec
}

type AlterSpec struct {
  PartitionName ColIdent
  PartitionDefinitions []*PartitionDefinition
}
```

Not critical but is more future proof, and we can keep DDL simple. Put all the formatting and such in Alter.